src/mmw/js/src/draw/views.js

```diff
@@ -208,31 +250,14 @@ function changeOutlineLayer(endpoint, tableId, model) {
     if (endpoint && tableId !== undefined) {
         var ol = new L.TileLayer(endpoint + '.png'),
             grid = new L.UtfGrid(endpoint + '.grid.json?callback={cb}', { resolution: 4 });
```

Comment:
Check out the option `maxRequests` https://github.com/danzel/Leaflet.utfgrid#other-options.

It indicates that it will usually prioritize the json requests over the image requests, potentially reducing the chances of this situation occurring.  Can you play with tweaking that option up to 8 and see how it affects the behavior?